Fortnox.NET/WebSockets/FortnoxWebSocketClient.cs

```diff
@@ -0,0 +1,273 @@
+            var AddTenantsResult = await SendAndRecieveOnce<WebSocketCommandResponse>(JsonConvert.SerializeObject(AddTenantsObject));
+
+            if (AddTenantsResult.Result.Equals("error"))
+            {
+                throw new WebSocketException("Unable to add tenants.");
+            }
+
+            if (_topics.Count <= 0)
+            {
+                throw new WebSocketException("No topics specified.");
+            }
+
+            var AddTopicObject = new { command = WebSocketCommands.AddTopics, topics = new List<object>() };
+            foreach(var topic in _topics)
+            {
+                AddTopicObject.topics.Add(new { topic = topic });
+            }
+
+            var AddTopicResult = await SendAndRecieveOnce<WebSocketCommandResponse>(JsonConvert.SerializeObject(AddTopicObject));
+            if (AddTopicResult.Result.Equals("error"))
+            {
+                throw new WebSocketException($"Error adding topic: {string.Join(",", AddTopicResult.InvalidTopics)}");
+            }
+
+            var SubscribeObject = new { command = WebSocketCommands.Subscribe };
+            var SubscribeResult = await SendAndRecieveOnce<WebSocketCommandResponse>(JsonConvert.SerializeObject(SubscribeObject));
+
+            if (!SubscribeResult.Result.Equals("ok"))
+            {
+                throw new WebSocketException("Failed to subscribe.");
+            }
+
+            return this;
+        }
+
+        /// <summary>
+        /// Adds the specified <see cref="WebSocketTopic"/> to the list of topics to subscribe to when calling <see cref="Connect"/>.
+        /// </summary>
+        /// <param name="topic">The target topic.</param>
+        /// <returns>The current instance of <see cref="FortnoxWebSocketClient"/></returns>
+        public FortnoxWebSocketClient AddTopic(WebSocketTopic topic)
+        {
+            Type type = topic.GetType();
+            FieldInfo fi = type.GetField(topic.ToString());
+            
+            WebSocketTopicStringValueAttribute attr = 
+                fi.GetCustomAttribute(typeof(WebSocketTopicStringValueAttribute), false) as WebSocketTopicStringValueAttribute;
+
+            this._topics.Add(attr.Value);
+
+            return this;
+        }
+
+        /// <summary>
+        /// Combines both <see cref="Connect"/> and <see cref="Listen(Func{ClientWebSocket, Task})"/> into a single function.
+        /// </summary>
+        /// <param name="callback">An asynchronous callback function that accepts a <see cref="System.Net.WebSockets.ClientWebSocket"/> in its parameters.</param>
+        /// <returns>An empty <see cref="Task"/></returns>
+        public async Task ConnectAndListen(Func<ClientWebSocket, Task> callback)
+        {
+            await Connect();
+            await callback.Invoke(this._webSocket);
+        }
+
+        /// <summary>
```

Comment:
@brokenprogrammer why is the commented out?

Reply:
I don't understand your question. This is documentation?